master/static/migrations/20230807002202_use-summary-metrics.tx.up.sql

```diff
@@ -0,0 +1,25 @@
+CREATE OR REPLACE FUNCTION autoupdate_exp_best_trial_metrics() RETURNS trigger AS $$
+BEGIN
+    WITH bt AS (SELECT id, best_validation_id FROM trials WHERE experiment_id = NEW.experiment_id ORDER BY searcher_metric_value_signed LIMIT 1)
+    UPDATE experiments SET best_trial_id = bt.id, 
+    validation_metrics = 
+    (
+        WITH metrics AS (
+            SELECT summary_metrics->'validation_metrics' AS jsonb, searcher_metric_value_signed=searcher_metric_value AS sign FROM trials WHERE id = bt.id
+        ), 
+        metrics_values AS (
+            SELECT jsonb_object_keys(jsonb) AS metrics_key, sign, jsonb -> jsonb_object_keys(jsonb) ->> 'min' AS min, jsonb -> jsonb_object_keys(jsonb) ->> 'max' AS max, jsonb -> jsonb_object_keys(jsonb) ->> 'type' AS type FROM metrics),
```

Comment:
what about `count` and `sum`? also, we probably need some documentation about summary metrics so this doesn't get missed if new metrics are added.

Reply:
also, this looks like it is applying the searcher metric sign to all metrics to determine the best? that isn't quite right, `searcher_metric_value_signed=searcher_metric_value` only applies as the sign for the `searcher_metric_value`.

i think we should have the experiment receive the best metrics according to the searcher metric value and just store than entire blob, from that trial. in which case this all gets much simpler, too.

Reply:
So smaller_is_better is only for searcher metrics? If that's the case, I don't think we can derive the "best" validation metrics values from summary metrics, we have to surface min, max, latest, mean for other validation metrics

Reply:
yes, right now, there is no way to know best for those metrics, only the user can decide by sort order.

Reply:
Got it. Upon discussing with product, we decided to surface min, max, latest, mean to validation metrics just as all other metrics.